test/unit/network/ConnectionRegistryTest.js

```diff
@@ -304,13 +182,12 @@ describe('ConnectionRegistryTest', function () {
 
         it('should return ClientOfflineError when reconnect mode is async, connection state is INITIALIZED_ON_CLUSTER '
             + 'and there are no connections', function () {
-                const connectionStrategyConfig = new ConnectionStrategyConfigImpl();
-                connectionStrategyConfig.reconnectMode = ReconnectMode.ASYNC;
-
                 const connectionRegistry = new ConnectionRegistryImpl(
-                    connectionStrategyConfig,
                     false,
-                    new RoundRobinLB()
+                    ReconnectMode.ASYNC,
+                    false,
+                    new RoundRobinLB(),
+                    {}
                 );
 
                 connectionRegistry.setConnectionState(connectionState.INITIALIZED_ON_CLUSTER);
```

Comment:
Can you add some tests for the new function `getConnectionForSql`, testing its behavior?

Reply:
Added some unit tests